filters/lifo/lifo.go

```diff
@@ -0,0 +1,177 @@
+package lifo
+
+import (
+	"net/http"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/zalando/skipper/filters"
+	"github.com/zalando/skipper/lifo"
+)
+
+// TODO: must be documented that it cannot be used together with the legacy shunting, meaning
+// that it's incompatible with MarkServed().
+
+type (
+	lifoSpec      struct{}
+	lifoGroupSpec struct{}
+)
+
+type lifoFilter struct {
+	config lifo.Config
+	stack  *lifo.Stack
+}
+
+type lifoGroupFilter struct {
+	name  string
+	stack *lifo.Stack
+}
+
+const (
+	lifoKey      = "lifo-done"
+	lifoGroupKey = "lifo-group-done"
+)
+
+const (
+	LIFOName      = "lifo"
+	LIFOGroupName = "lifoGroup"
+)
+
+func NewLIFO() filters.Spec {
+	return &lifoSpec{}
+}
+
+func NewLIFOGroup() filters.Spec {
+	return &lifoGroupSpec{}
+}
+
+func intArg(a interface{}) (int, error) {
+	switch v := a.(type) {
+	case int:
+		return v, nil
+	case float64:
+		return int(v), nil
+	default:
+		return 0, filters.ErrInvalidFilterParameters
+	}
+}
+
+func durationArg(a interface{}) (time.Duration, error) {
+	switch v := a.(type) {
+	case string:
+		return time.ParseDuration(v)
+	default:
+		return 0, filters.ErrInvalidFilterParameters
+	}
+}
+
+func (s *lifoSpec) Name() string { return LIFOName }
+
+func (s *lifoSpec) CreateFilter(args []interface{}) (filters.Filter, error) {
+	var (
+		l   lifoFilter
+		err error
+	)
+
+	if len(args) > 0 {
+		if l.config.MaxConcurrency, err = intArg(args[0]); err != nil {
+			return nil, err
+		}
+	}
+
+	if len(args) > 1 {
+		if l.config.MaxStackSize, err = intArg(args[1]); err != nil {
+			return nil, err
+		}
+	}
+
+	if len(args) > 2 {
+		if l.config.Timeout, err = durationArg(args[2]); err != nil {
+			return nil, err
+		}
+	}
+
+	if len(args) > 3 {
+		return nil, filters.ErrInvalidFilterParameters
+	}
+
```

Comment:
What do you think about validating sane values for the parameters? At least maxstacksize = 0 would seem unreasonable

Reply:
Valid point. Currently this relies on the very general rules from here: https://godoc.org/github.com/aryszka/jobstack#Options . Which means defaults: concurrency=1, stacksize=infinite, timeout=infinite, based on the zero-one-infinite principle. May not make sense for our use case in Skipper, I'll think about it.

Reply:
I also realize that it may not be clear, and the naming can be misleading. Currently, the total number of requests = requests in the stack + requests being handled, which means total max = max stack + max concurrency. It requires better naming.

Reply:
I added this to CreateFilter docs

Reply:
I added checks with set to min value and doc for this.